docs/contract_types.rst

```diff
@@ -324,6 +324,65 @@ Note: you are not allowed to mix different kinds of separators on the same line.
         mypackage.low
 
 
+Protected modules
+-----------------
+
+*Type name:* ``protected``
+
+Protected contracts prevent certain modules from being directly imported, except by modules in an allow-list. 
+By default, descendants of each module will be checked too.
+
+For example, if blue is protected, and green is the only module in the allow list,
+then no module other than green (and its descendants) will be allowed to import blue (and its descendants) directly.
+
+**Examples:**
+
+.. code-block:: ini
+
+    [importlinter]
+    root_package = mypackage
+
+    [importlinter:contract:my-simple-protected-contract]
+    name = My simple protected contract
+    type = protected
+    protected_modules =
+        mypackage.one
+    allowed_importers =
+        mypackage.two
+        mypackage.three.blue
+    
+.. code-block:: ini
+
+    [importlinter]
+    root_package = mypackage
+
+    [importlinter:contract:my-protected-contract]
+    name = My protected contract
```

Comment:
Suggest calling this `Models can only be imported by colors` or `Contract with wildcards` (not sure which you think is better).